iroh-util/src/lock.rs

```diff
@@ -37,21 +40,62 @@ impl ProgramLock {
 
     /// Try to acquire a lock for this program.
     pub fn acquire(&mut self) -> Result<()> {
-        let file = Rc::new(File::create(&self.path)?);
+        let mut file = File::create(&self.path)?;
+        file.write_u32::<LittleEndian>(process::id())?;
+        let file = Rc::new(file);
 
         file_guard::lock(file, Lock::Exclusive, 0, 1)
             .map(|lock| self.lock = Some(lock))
             .map_err(|err| err.into())
     }
 }
 
+/// Attempt to remove a stray lock file that wasn't cleaned up, returns true
+/// if a lock is successfully deleted, and will only attempt to delete if the
+/// lock is not currently held
+pub fn try_cleanup_dead_lock(prog_name: &str) -> Result<bool> {
```

Comment:
This needs a test, and I need to review it when I'm not braindead, please scrutinize.